Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,189 @@
+                        var guardian = EntityManager.SpawnEntity(component.GuardianType, hostcomp.Owner.Transform.Coordinates);
+                        hostcomp.GuardianContainer.Insert(guardian);
+                        hostcomp.HostedGuardian = guardian.Uid;
+                        //Takes the guardian component on the supposed guardian entity and fills out it's synced parametres
+                        if (guardian.TryGetComponent<GuardianComponent>(out GuardianComponent? guardiancomp))
+                        {
+                            guardiancomp.Host = args.User.Uid;
+                            //Grant the user the recall action and notify them
+                            action.Grant(ActionType.ManifestGuardian);
+                            SoundSystem.Play(Filter.Entities(uid), "/Audio/Effects/guardian_inject.ogg", uid);
+                            _popupSystem.PopupCoordinates(Loc.GetString("guardian-created"), hostcomp.Owner.Transform.Coordinates, Filter.Entities(args.User.Uid));
+                            //Exhaust the activator
+                            component.Used = true;
+                        }
+                    }
+                    else
+                    {
+                        _popupSystem.PopupCoordinates(Loc.GetString("guardian-activator-empty-invalid-creaton"), args.User.Transform.Coordinates, Filter.Entities(args.User.Uid));
+                    }
+                }
+                else
+                {
+                    _popupSystem.PopupCoordinates(Loc.GetString("guardian-no-actions-invalid-creation"), args.User.Transform.Coordinates, Filter.Entities(args.User.Uid));
+                }
+            }
+            else
+            {
+                _popupSystem.PopupCoordinates(Loc.GetString("guardian-already-present-invalid-creation"), args.User.Transform.Coordinates, Filter.Entities(args.User.Uid));
+            }
+        }
+
+        /// <summary>
+        /// Triggers when the host recives damage which puts the host in either critical or killed state
+        /// </summary>
+        private void OnHostStateChange(EntityUid uid, GuardianHostComponent component, MobStateChangedEvent args)
+        {
+                if (args.State.IsCritical())
+                {
+                    _popupSystem.PopupCoordinates(Loc.GetString("guardian-host-critical-warn"), EntityManager.GetComponent<TransformComponent>(uid).Coordinates, Filter.Entities(component.HostedGuardian));
+                    SoundSystem.Play(Filter.Local(), "/Audio/Effects/guardian_warn.ogg", component.HostedGuardian);
+                }
+                else if (args.State.IsDead())
+                {
+                    SoundSystem.Play(Filter.Pvs(component.HostedGuardian), "/Audio/Voice/Human/malescream_guardian.ogg", uid, AudioHelpers.WithVariation(0.20f));
+                    EntityManager.QueueDeleteEntity(component.HostedGuardian);
+                    EntityManager.QueueDeleteEntity(uid);
+                }
```

Comment:
Should the person really get deleted?

Reply:
I thought we solved this, yes, that's what the consensus is until a PR with proper dust kill comes around